app/v1/models/request/submit/SubmitRequestBody.scala

```diff
@@ -16,23 +16,42 @@
 
 package v1.models.request.submit
 
-import play.api.libs.json.{Json, OWrites, Reads}
+import play.api.libs.functional.syntax._
+import play.api.libs.json._
+
+case class SubmitRequestBody(periodKey: Option[String],
+                             vatDueSales: Option[BigDecimal],
+                             vatDueAcquisitions: Option[BigDecimal],
+                             vatDueTotal: Option[BigDecimal],
+                             vatReclaimedCurrPeriod: Option[BigDecimal],
+                             vatDueNet: Option[BigDecimal],
+                             totalValueSalesExVAT: Option[BigDecimal],
+                             totalValuePurchasesExVAT: Option[BigDecimal],
+                             totalValueGoodsSuppliedExVAT: Option[BigDecimal],
+                             totalAllAcquisitionsExVAT: Option[BigDecimal],
+                             finalised: Option[Boolean],
+                             receivedAt: Option[String],
+                             agentReference: Option[String]) {
+}
 
-case class SubmitRequestBody(periodKey: String,
-                             vatDueSales: BigDecimal,
-                             vatDueAcquisitions: BigDecimal,
-                             totalVatDue: BigDecimal,
-                             vatReclaimedCurrPeriod: BigDecimal,
-                             netVatDue: BigDecimal,
-                             totalValueSalesExVAT: BigDecimal,
-                             totalValuePurchasesExVAT: BigDecimal,
-                             totalValueGoodsSuppliedExVAT: BigDecimal,
-                             totalAcquisitionsExVAT: BigDecimal,
-                             finalised: Boolean)
 
 object SubmitRequestBody {
 
-  implicit val reads: Reads[SubmitRequestBody] = Json.reads[SubmitRequestBody]
-  implicit val writes: OWrites[SubmitRequestBody] = Json.writes[SubmitRequestBody]
+  implicit val reads: Reads[SubmitRequestBody] = (
+    (JsPath \ "periodKey").readNullable[String] and
+      (JsPath \ "vatDueSales").readNullable[BigDecimal] and
+      (JsPath \ "vatDueAcquisitions").readNullable[BigDecimal] and
+      (JsPath \ "totalVatDue").readNullable[BigDecimal] and
+      (JsPath \ "vatReclaimedCurrPeriod").readNullable[BigDecimal] and
+      (JsPath \ "netVatDue").readNullable[BigDecimal] and
+      (JsPath \ "totalValueSalesExVAT").readNullable[BigDecimal] and
+      (JsPath \ "totalValuePurchasesExVAT").readNullable[BigDecimal] and
+      (JsPath \ "totalValueGoodsSuppliedExVAT").readNullable[BigDecimal] and
+      (JsPath \ "totalAcquisitionsExVAT").readNullable[BigDecimal] and
+      (JsPath \ "finalised").readNullable[Boolean] and
+      (JsPath \ "receivedAt").readNullable[String] and
+      (JsPath \ "agentReference").readNullable[String]
```

Comment:
These fields are not supplied in the vendor request so putting them in the reads like this doesn't actually do anything.